server/server.go

```diff
@@ -462,7 +463,7 @@ func (s *Server) setupRelayer() error {
 		s.config.JSONRPC.JSONRPCAddr.String(),
 		ethgo.Address(contracts.StateReceiverContract),
 		s.logger.Named("relayer"),
-		wallet.NewEcdsaSigner(wallet.NewKey(account)),
+		wallet.NewEcdsaSigner(wallet.NewKey(account, bls.DomainCheckpointManager)),
```

Comment:
Is it correct using `DomainCheckpointManager`, since this transaction is being sent to `StateReceiver` SC, which is deployed on L2?

Generally speaking, when we are using `DomainCheckpointManager` and when `DomainValidatorSet`?

Reply:
They are calling `(bool callSuccess, bytes memory returnData) = VALIDATOR_PKCHECK_PRECOMPILE.staticcall{` and there we use  `DomainCheckpointManager`